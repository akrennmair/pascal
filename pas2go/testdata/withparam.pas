program test;

type foox = record
			a : integer;
			b : real
		end;
	fooy = record
		z : record
			c : integer
		end
	end;

var xx : foox;
	yy : fooy;

procedure quux(x : foox; y : fooy);
begin
	with x, y do
	begin
		a := 42;
		b := 23.5;
		writeln('a = ', a);
		writeln('b = ', b);
		with z do
		begin
			c := 9001;
			writeln('c = ', c)
		end
	end
end;

begin
	quux(xx, yy)
end.
