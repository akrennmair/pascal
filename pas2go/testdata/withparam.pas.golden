package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	type (
		foox struct {
			a int
			b float64
		}
		fooy struct {
			z struct {
				c int
			}
		}
	)

	var (
		xx foox
		yy fooy
	)
	_ = xx
	_ = yy

	var quux func(x foox, y fooy)
	quux = func(x foox, y fooy) {

		x.a = 42
		x.b = 23.5e0
		system.Writeln("a = ", x.a)
		system.Writeln("b = ", x.b)

		y.z.c = 9001
		system.Writeln("c = ", y.z.c)
		return
	}

	quux(xx, yy)
}
